### PR TITLE
Improve editor property name extraction

### PR DESCRIPTION
**Pull request by timothyqiu**
* A group will remove prefix from property paths following it when displaying. This should be done when extracting too.
    * We can't let the translators translate the full path and then remove prefix from the translated message. Word orders are not always kept after translation.
    * A `ExtractType` enum is added in `extract.py`. We now distinguish what a pattern extracts with the enum instead of a simple boolean.
* Fixes a regex bug where properties using default hints are not extracted
* Adds more missing capitalization remaps.
* Gets our feet wet about translation context by adding context to "Project Manager" strings.

### CODE REVIEW

File: editor/project_manager.cpp [hunk 1]
```diff
@@ -2535,7 +2535,7 @@ ProjectManager::ProjectManager() {
 	}
 
 	// TRANSLATORS: This refers to the application where users manage their Godot projects.
-	DisplayServer::get_singleton()->window_set_title(VERSION_NAME + String(" - ") + TTR("Project Manager"));
+	DisplayServer::get_singleton()->window_set_title(VERSION_NAME + String(" - ") + TTR("Project Manager", "Application"));
```
Comment (akien-mga) [inline editor/project_manager.cpp:2538]: That one doesn't seem to work (with the `3.x` version):
```diff
@@ -5002,8 +5399,7 @@ msgstr ""
 msgid "Editor SSL Certificates"
 msgstr ""
 
-#. TRANSLATORS: This refers to the application where users manage their Godot projects.
-#: editor/editor_settings.cpp editor/project_manager.cpp
+#: editor/editor_settings.cpp
 msgid "Project Manager"
 msgstr ""
 
```

Comment (akien-mga) [reply editor/project_manager.cpp:2538]: Or more like, there's another one in `editor/editor_settings.cpp` to change.

Comment (timothyqiu) [reply editor/project_manager.cpp:2538]: That's part of a property path... I was about to add context support for property path :)

But I think it'll be delayed after the inspector toggle. I'll remove the context for this one.

**Comment by akien-mga** [comment]
Some more remaps that I found reviewing the POT diff:
```diff
diff --git a/editor/editor_property_name_processor.cpp b/editor/editor_property_name_processor.cpp
index c67db94ab8..49f2f21095 100644
--- a/editor/editor_property_name_processor.cpp
+++ b/editor/editor_property_name_processor.cpp
@@ -83,6 +83,8 @@ EditorPropertyNameProcessor::EditorPropertyNameProcessor() {
 	capitalize_string_remaps["aabb"] = "AABB";
 	capitalize_string_remaps["adb"] = "ADB";
 	capitalize_string_remaps["ao"] = "AO";
+	capitalize_string_remaps["arvr"] = "ARVR";
+	capitalize_string_remaps["bg"] = "BG";
 	capitalize_string_remaps["bp"] = "BP";
 	capitalize_string_remaps["bpc"] = "BPC";
 	capitalize_string_remaps["bptc"] = "BPTC";
@@ -94,6 +96,7 @@ EditorPropertyNameProcessor::EditorPropertyNameProcessor() {
 	capitalize_string_remaps["db"] = "dB";
 	capitalize_string_remaps["dof"] = "DoF";
 	capitalize_string_remaps["dpi"] = "DPI";
+	capitalize_string_remaps["dtls"] = "DTLS";
 	capitalize_string_remaps["etc"] = "ETC";
 	capitalize_string_remaps["etc2"] = "ETC2";
 	capitalize_string_remaps["fft"] = "FFT";
@@ -113,11 +116,15 @@ EditorPropertyNameProcessor::EditorPropertyNameProcessor() {
 	capitalize_string_remaps["hdr"] = "HDR";
 	capitalize_string_remaps["hidpi"] = "hiDPI";
 	capitalize_string_remaps["hipass"] = "High-pass";
+	capitalize_string_remaps["hsv"] = "HSV";
 	capitalize_string_remaps["http"] = "HTTP";
+	capitalize_string_remaps["id"] = "ID";
 	capitalize_string_remaps["ik"] = "IK";
 	capitalize_string_remaps["igd"] = "IGD";
 	capitalize_string_remaps["ios"] = "iOS";
 	capitalize_string_remaps["iod"] = "IOD";
+	capitalize_string_remaps["ip"] = "IP";
+	capitalize_string_remaps["ipv6"] = "IPv6";
 	capitalize_string_remaps["k1"] = "K1";
 	capitalize_string_remaps["k2"] = "K2";
 	capitalize_string_remaps["kb"] = "(KB)"; // Unit.
@@ -126,7 +133,8 @@ EditorPropertyNameProcessor::EditorPropertyNameProcessor() {
 	capitalize_string_remaps["macos"] = "macOS";
 	capitalize_string_remaps["mb"] = "(MB)"; // Unit.
 	capitalize_string_remaps["ms"] = "(ms)"; // Unit
-	capitalize_string_remaps["msec"] = "(msec)"; // Unit.
+	// Not used for now as AudioEffectReverb has a `msec` property.
+	//capitalize_string_remaps["msec"] = "(msec)"; // Unit.
 	capitalize_string_remaps["msaa"] = "MSAA";
 	capitalize_string_remaps["normalmap"] = "Normal Map";
 	capitalize_string_remaps["opengl"] = "OpenGL";
@@ -137,11 +145,13 @@ EditorPropertyNameProcessor::EditorPropertyNameProcessor() {
 	capitalize_string_remaps["pvs"] = "PVS";
 	capitalize_string_remaps["pvrtc"] = "PVRTC";
 	capitalize_string_remaps["rid"] = "RID";
+	capitalize_string_remaps["rmb"] = "RMB";
 	capitalize_string_remaps["rpc"] = "RPC";
 	capitalize_string_remaps["s3tc"] = "S3TC";
 	capitalize_string_remaps["sdf"] = "SDF";
 	capitalize_string_remaps["sdfgi"] = "SDFGI";
 	capitalize_string_remaps["sdk"] = "SDK";
+	capitalize_string_remaps["sec"] = "(sec)"; // Unit.
 	capitalize_string_remaps["ssao"] = "SSAO";
 	capitalize_string_remaps["ssh"] = "SSH";
 	capitalize_string_remaps["ssil"] = "SSIL";
```

For 4.0 I'll rework the use of `s` or `sec` as suffix in some properties, to at least make it consistent (and probably rename the `predelay_msec` in `AudioEffectReverb` to `predelay_duration_ms` or something).

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**